File: hellocharts/src/main/java/net/raquezha/lecho/hellocharts/animation/ChartViewportAnimator.java
package net.raquezha.lecho.hellocharts.animation;

import net.raquezha.lecho.hellocharts.model.Viewport;

public interface ChartViewportAnimator {

    int FAST_ANIMATION_DURATION = 300;

    void startAnimation(Viewport startViewport, Viewport targetViewport);

    void startAnimation(Viewport startViewport, Viewport targetViewport, long duration);

    void cancelAnimation();

    @SuppressWarnings("unused")
    boolean isAnimationStarted();

    void setChartAnimationListener(ChartAnimationListener animationListener);

}

File: hellocharts/src/main/java/lecho/lib/hellocharts/animation/ChartViewportAnimator.kt
package lecho.lib.hellocharts.animation

import lecho.lib.hellocharts.model.Viewport

interface ChartViewportAnimator {
    fun startAnimation(startViewport: Viewport, targetViewport: Viewport)
    fun startAnimation(startViewport: Viewport, targetViewport: Viewport, duration: Long)
    fun cancelAnimation()
    val isAnimationStarted: Boolean
    fun setChartAnimationListener(animationListener: ChartAnimationListener?)

    companion object {
        const val FAST_ANIMATION_DURATION = 300
    }
}